avocado/utils/data_structures.py

```diff
@@ -218,3 +218,26 @@ def __del__(self):
                   to be executed!
         """
         self.run()
+
+
+def time_to_seconds(time):
+    """
+    Convert time in minutes, hours and days to seconds.
+    :param time: Time including the unit (i.e. '10d')
+    """
+    units = {'s': 1, 'm': 60, 'h': 3600, 'd': 86400}
+    if time is not None:
+        try:
+            unit = time[-1].lower()
+            if unit in units:
+                mult = units[unit]
+                seconds = int(time[:-1]) * mult
+            else:
+                seconds = int(time)
+        except (ValueError, TypeError) as e:
+            raise Exception("Invalid value '%s' for time. Use an integer "
```

Comment:
Using the `Exception` is not much recommended as most of the exceptions inherits from it. You should chose the most appropriate one or create new one in case it's needed. (I'd go with `ValueError` as there is an invalid value).